"""This is a program abour how to use try, exeption, raise"""

def divisor(num):
    try:
        if num < 0:
            raise ValueError("put a number")
        divisors =[i for i in range(1, num + 1) if num % i == 0]
        return divisors
    except ValueError as ve:
        print(ve)
        return False


def run():
    try: 
        num = int(input(f'put a number: '))      
        print(divisor(num))
        print("program finished")
    except ValueError:
        print("put a number")
    
if __name__ =='__main__':
    run()